tao_compiler/mlir/disc/transforms/disc_dot_merge.cc

```diff
@@ -190,6 +236,361 @@ void BuildDotClusters(Block* block,
   }
 }
 
+class DotShareOperandMergeConverter {
+ public:
+  DotShareOperandMergeConverter(FuncOp func) : func_(func){};
+  bool run();
+
+ public:
+  enum ShareType : int32_t { LEFT = 0, RIGHT = 1 };
+
+  struct DotShareInfo {
+    ShareType share_type;
+    Value share_operand;
+    int64_t lhs_contracting_dim;
+    int64_t rhs_contracting_dim;
```

Comment:
It is not enough to log only contracting dims. It requires all dims the same layout. Please use the overall dim-numbers and add UT to test different layouts.